eis_toolkit/prediction/weights_of_evidence.py

```diff
@@ -0,0 +1,255 @@
+from numbers import Number
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import rasterio
+from beartype import beartype
+from beartype.typing import List, Literal, Optional, Sequence, Tuple
+
+from eis_toolkit import exceptions
+from eis_toolkit.vector_processing.rasterize_vector import rasterize_vector
+
+
+def _read_and_preprocess_evidence(raster: rasterio.io.DatasetReader, nodata: Optional[Number] = None) -> np.ndarray:
+    """Read raster data and handle NoData values."""
+
+    array = np.array(raster.read(1), dtype=np.float32)
```

Comment:
```suggestion
def _read_and_preprocess_evidence(raster: rasterio.io.DatasetReader, nodata: Optional[Number] = None, band:int = 1) -> np.ndarray:
    """Read raster data and handle NoData values."""

    array = np.array(raster.read(band), dtype=np.float32)
```

Assumes that 1 is the correct band. Should at least add option to override.

Reply:
Good catch. Modified according to your suggestion, and should probably modify to work with multiband rasters in a later stage as well.